aii-ks/src/main/perl/ks.pm

```diff
@@ -427,12 +427,31 @@ timezone --utc $tree->{timezone}$ntp_servers
 rootpw --iscrypted $tree->{rootpw}
 EOF
 
+    my $repos = get_repos($config);
+    # error reported in get_repos
+    return if ! $repos;
+
     foreach my $url (@{$tree->{repo} || []}) {
         if ($url =~ m/^@(.+)$/) {
+            # find at least one repo with matching name
+            my $pattern = $1;
+            my @matches = grep {m/$pattern/} sort keys %$repos;
```

Comment:
I'd really prefer if the pattern was anchored on both ends by default. I also think interpreting the value as a glob rather than as a regular expression would be simpler to use. I can see being able to write `@productname*` being useful to select all repositories related to a product, but I'd really not want to see templates/repository names which would require more complex matching :-)

Reply:
good idea, i switch to using a glob